kmath-commons-rng-part/src/commonMain/kotlin/scientifik/commons/rng/sampling/SharedStateSampler.kt

```diff
@@ -0,0 +1,7 @@
+package scientifik.commons.rng.sampling
+
+import scientifik.commons.rng.UniformRandomProvider
+
+interface SharedStateSampler<R> {
```

Comment:
This should be removed. Current ideology is to set a generator once during chain creation. If we want to change the API, we need to discuss it. We can still use state for forks.